cpp/arcticdb/util/hash.hpp

```diff
@@ -18,23 +18,17 @@
 
 namespace arcticdb {
 
-using HashedValue = XXH64_hash_t;
-constexpr std::size_t DEFAULT_SEED = 0x42;
-
-template<class T, std::size_t seed = DEFAULT_SEED>
-HashedValue hash(T *d, std::size_t count) {
-    return XXH64(reinterpret_cast<const void *>(d), count * sizeof(T), seed);
+template<class T>
+inline size_t hash(T d) {
+    return std::hash<T>{}(d);
 }
 
-// size argument to XXH64 being compile-time constant improves performance
-template<class T, std::size_t seed = DEFAULT_SEED>
-HashedValue hash(T *d) {
-    return XXH64(reinterpret_cast<const void *>(d), sizeof(T), seed);
+inline size_t hash(std::string_view sv) {
+    return std::hash<std::string_view>{}(sv);
 }
 
-inline HashedValue hash(std::string_view sv) {
-    return hash(sv.data(), sv.size());
-}
+using HashedValue = XXH64_hash_t;
+constexpr std::size_t DEFAULT_SEED = 0x42;
```

Comment:
It looks like after this change `DEFAULT_SEED` is used only in `HashAccum`. Can we move it inside the class in that case?